airflow/cli/commands/task_command.py

```diff
@@ -199,6 +199,8 @@ def _get_ti_db_access(
             )
         # TODO: Validate map_index is in range?
         ti = TaskInstance(task, run_id=dag_run.run_id, map_index=map_index)
+        if create_if_necessary == "db":
+            session.add(ti)
```

Comment:
Would this not just delay this until when we need to handle `create_if_necessary == "memory"`?

Reply:
Yeah that check looks much sensible to me.